### ISSUE TITLE
Programmatic access to completion URLs

### ISSUE THREAD

**Issue by nqn** [log10-io/log10#22]
Completion URLs can be printed using the `DEBUG_` mode when wrapping openai, like so:

```
log10(openai, DEBUG_=True)
```

But is a bit verbose, and not easily grabbed programmatically

**Comment by treppers** [comment]
This actually is a bit tricky for me to parse, because I'm currently grabbing std_out only for langchain's output which happens well after the openai calls.

Ideally I could grab a URL (ideally the last session, but also the last chain works) from memory after the chain has been ran.

here's how i'm currently running the top-level agent.
`output_buffer = StringIO()
    with log10_session():
        try:
            with contextlib.redirect_stdout(output_buffer):
                chainResult = chain({"human_input": inputs.question}, return_only_outputs=True)
            
            logging.info("chain successfully executed")
        except Exception as e:
            chainResult = output_buffer.getvalue()

            logging.error("Execution Result: " + chainResult)
            logging.error("Exception: " + str(e))
            logging.error(traceback.format_exc())
            return {"status": "error", "error": str(e)}
    
    chainOutput = output_buffer.getvalue()`

however rather than parsing it from std_out, would great to just run `log10_last_completion_url()` and get that.

**Comment by nqn** [comment]
Great point! I think we can solve this by providing access to a session object, like

```
with log10_session() as session:
  ...
  session.completion_url()
```

Would that work?

**Comment by treppers** [comment]
that would be a huge help.

currently we have multiple users banging on staging and/or having a series of q&a with the bot it's hard to find the completion id from the list.  so putting this link directly to the front-end will be a huge unlock for us and our workflow.

**Comment by arjunbansal** [comment]
Fixed via #23
